flutter_screen_recording/lib/mixin/record_screen_mixin.dart

```diff
@@ -0,0 +1,128 @@
+
+  final recordingStatusStreamController = StreamController<bool>()..add(false);
+  final recordPauseStatusStreamController = StreamController<bool>()
+    ..add(false);
+
+  Stream<bool> recordingStatusStream;
+  Stream<bool> recordPauseStatusStream;
+
+  bool recordingScreen = false;
+
+  void onInitScreenRecord() {
+    recordingStatusStream =
+        recordingStatusStreamController.stream.asBroadcastStream();
+    recordPauseStatusStream =
+        recordPauseStatusStreamController.stream.asBroadcastStream();
+  }
+
+  void startRecordScreen(String fileName) async {
+    final result = await FlutterScreenRecording.startRecordScreenAndAudio(
+      fileName,
+      recordSystemAudio: true,
+      disableUserAudio: true,
+      onStop: onStopRecording,
+    );
+    recordingScreen = result;
+    recordingStatusStreamController.add(result);
+    addCurrentAudioTracksToRecordScreenStream();
+  }
+
+  void pauseRecordScreen() {
+    final pauseStatus = FlutterScreenRecording.pauseRecordScreen();
+    recordPauseStatusStreamController.add(pauseStatus);
+  }
+
+  void resumeRecordScreen() {
+    final pauseStatus = !FlutterScreenRecording.resumeRecordScreen();
+    recordPauseStatusStreamController.add(pauseStatus);
+  }
+
+  void stopRecordScreen() {
+    FlutterScreenRecording.stopRecordScreen;
+  }
+
+  void onStopRecording(result) {
+    recordingScreen = false;
+    recordingStatusStreamController.add(false);
+    participantsAudioSourceNodeIds.clear();
+  }
+
+  void storeAudioTrackReference(String uid, dynamic audioStream) {
+    participantsAudioTracks[uid] = audioStream as MediaStream;
+  }
+
+  void removeAudioTrackReference(String uid) {
+    participantsAudioTracks.remove(uid);
+  }
+
+  void addAudioTrackToScreenRecord(String uid, dynamic audioStream) {
+    final result =
+        FlutterScreenRecording.addAudioTrack(audioStream as MediaStream);
+
+    participantsAudioSourceNodeIds[uid] = result;
+  }
+
+  void removeAudioTrackFromScreenRecord(dynamic uid) {
+    final mediaStreamAudioSourceNodeId = participantsAudioSourceNodeIds[uid];
+    if (mediaStreamAudioSourceNodeId != null) {
+      FlutterScreenRecording.removeAudioTrack(mediaStreamAudioSourceNodeId);
+    }
+    participantsAudioSourceNodeIds.remove(uid);
+  }
+
+  void addCurrentAudioTracksToRecordScreenStream() {
+    for (final uid in participantsAudioTracks.keys) {
+      final audioStream = participantsAudioTracks[uid];
+      if (audioStream != null) {
+        final result = FlutterScreenRecording.addAudioTrack(audioStream);
+
+        participantsAudioSourceNodeIds[uid] = result;
+      }
+    }
+  }
+
+  void onDisposeScreenRecord() {
+    recordingScreen = false;
+    participantsAudioTracks.clear();
+    participantsAudioSourceNodeIds.clear();
+    recordingStatusStreamController.close();
+    recordPauseStatusStreamController.close();
+  }
+
+  Widget buildRecordingPauseStatusContainer(
+    BuildContext context,
```

Comment:
this does not make sense when we input context and return context in builder

Reply:
- Any parent (input context)
      - StreamBuilder 
             - builder func (stream builder context)

Input a context to remind that StreamBuilder widget will build on a context(to know where it is in the element tree).
Output context in the builder func is the context from the StreamBuilder - not from the inpur context.

Reply:
but you don't use it. And StreamBuilder is a widget, we can implement when we use, no need to have these function

Reply:
and don't put ui in logic class